gpkit/solution_array.py

```diff
@@ -592,6 +593,23 @@ def savetxt(self, filename="solution.txt", *, printmodel=True, **kwargs):
                 f.write(self.modelstr + "\n")
             f.write(self.table(**kwargs))
 
+    def savejson(self, filename="solution.json"):
+        "Saves solution table as a json file"
+        sol_dict = {}
+        # get list of variables
+        data = self["variables"]
+        # add appropriate data for each variable to the dictionary
+        for i, (k, v) in enumerate(data.items()):
+            key = str(k.name)
```

Comment:
This should probably be `str(k)` within a `for key in self.name_collision_varkeys():  key.descr["necessarylineage"] = True/False` environment, as in `def tables(...)` below.